content/guides/clj_datatype_constructs.adoc

```diff
@@ -0,0 +1,162 @@
+= Understanding Clojure's Datatype Constructs
+Ikuru Kanuma
+2017-07-20
+:type: guides
+:toc: macro
+:icons: font
+
+ifdef::env-github,env-browser[:outfilesuffix: .adoc]
+
+== Goals of this guide
+
+Clojure supports several constructs for speaking to the Java world
+and/or creating types for polymorphic dispatch. +
+Because these constructs have overlapping capabilities, it may be confusing to know which construct to use at a given situation. +
+This guide clarifies what each construct is good at, while presenting minimal usage examples.
+
+== Proxy a Java class and/or Interfaces
```

Comment:
Proxies should probably be the least frequently used so I don't like starting this guide with it. Should move to the end.